File: ktor/src/main/kotlin/dev/nonava/techstacks/ktor/greeting/infrastructure/module/GreetingModule.kt
/*
 * Copyright 2021-2022 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package dev.nonava.techstacks.ktor.greeting.infrastructure.module

import dev.nonava.techstacks.ktor.greeting.application.service.GreetingService
import org.koin.core.module.Module
import org.koin.dsl.module

val greetingModule: Module = module {
    single { GreetingService() }
}

File: nest/src/greeting/infrastructure/module/greeting.module.ts
/*
 * Copyright 2021-2022 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

import {Module} from "@nestjs/common";
import {GreetingController} from "../controller/greeting.controller";
import {GreetingService} from "../../application/service/greeting.service";

@Module({
    imports: [],
    controllers: [GreetingController],
    providers: [GreetingService]
})
export class GreetingModule {
}
